### DISCUSSION TITLE
Searching embeddings locally

### DISCUSSION THREAD

**Discussion by nightfall708** [supabase/supabase#20276] [Questions]
Hi,

I'm kinda new to AI and Supabase. I learned that Supabase supports embeddings and pgvector. Was wondering if I have an iOS Swift app, is it possible to generate embeddings and do a semantic search locally.

Thank you..!

**Comment by alt-glitch** [comment]
Hey @nightfall708 
Given Supabase is a Database-as-a-Service tool, it can't help you in generating embeddings and storing them locally (on device) for your iOS application.

But you can use Supabase to store and query embeddings you generate from OpenAI or other models.

**Reply by nightfall708** [reply]
Thanks!
Can you refer me to an example on storing and querying embeddings?

**Reply by alt-glitch** [reply]
[This doc](https://supabase.com/docs/guides/ai/vector-columns) here talks about creating vector columns, storing and querying them.

If you've used HuggingFace Transformers, then [this doc](https://supabase.com/docs/guides/ai/quickstarts/generate-text-embeddings) shows how to use Transformers.js with Supabase.

I hope this was helpful enough :)

**Reply by nightfall708** [reply]
Can the embeddings be searched for locally?

**Reply by alt-glitch** [reply]
Embedding search can only be done through the DB, where they're stored. [This](https://github.com/supabase-community/nextjs-openai-doc-search?tab=readme-ov-file#-runtime) is an example of how embedding search and querying could be done.
